internal/prometheus/prometheus.go

```diff
@@ -116,12 +120,21 @@ func (p *Provider) All() map[string]interface{} {
 	}
 
 	for _, f := range families {
-		all[f.GetName()] = f
+		all[f.GetName()] = wrap{family: f}
 	}
 
 	return all
 }
 
+type wrap struct{ family proto.Message }
+
+var marshaler = jsonpb.Marshaler{}
+
+func (w wrap) MarshalJSON() ([]byte, error) {
```

Comment:
trying to have the reflect machinery underneath encoding/json pick up that these metrics should use a different marshalling function to generate json.